test/RpcCompatibility.js

```diff
@@ -484,4 +484,21 @@ describe('RpcCompatibility', () => {
             await checkReceipt(receipt, newValue);
         });
     });
+
+    it('should throw error when querying eth_getLogs with a future blockHash', async () => {
+        const currentBlockNumber = await provider.getBlockNumber();
+        const futureBlockNumber = currentBlockNumber + 1000;
+
+        const fakeFutureBlockHash = ethers.utils.hexZeroPad(ethers.utils.hexlify(futureBlockNumber), 32);
+
+        const params = [
+            {
+                blockHash: fakeFutureBlockHash,
+            },
+        ];
+
+        if (!isHardhat) {
```

Comment:
What does hardhat do? Can its behavior be tuned by the hardhat config to match actual chains?